src/Formats/ProtobufSerializer.cpp

```diff
@@ -3353,20 +3353,63 @@ namespace
                         /// Now we have up to `nested_message_column_names.size()` columns
                         /// which can be serialized as a nested message.
 
-                        /// We will try to serialize those columns as one nested message,
-                        /// then, if failed, as an array of nested messages (on condition if those columns are array).
-                        bool has_fallback_to_array_of_nested_messages = false;
+                        bool nested_all_arrays = false ;
+                        bool nested_some_arrays = false ;
```

Comment:
I need to think a bit about the variables naming. This one in particular might be better as `has_some_nested_arrays`

Reply:
The other one could be `all_nested_columns_are_arrays`

Reply:
renamed to 
- `repeated_field_matching_nested_columns_are_all_arrays`
- `repeated_field_matching_nested_columns_have_some_arrays`

for even more explicitness :pray: